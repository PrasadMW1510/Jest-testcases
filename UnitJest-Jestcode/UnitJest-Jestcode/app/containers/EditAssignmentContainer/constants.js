/*
 *
 * EditAssignmentContainer constants
 *
 */

export const GET_ASSIGNMENT_REQUEST = 'app/EditAssignmentContainer/GET_ASSIGNMENT_REQUEST';
export const GET_ASSIGNMENT_SUCCESS = 'app/EditAssignmentContainer/GET_ASSIGNMENT_SUCCESS';
export const GET_STUDENTDETAIL_SUCCESS = 'app/EditAssignmentContainer/GET_STUDENTDETAIL_SUCCESS';
export const GET_STUDENT_DETAILS = 'app/EditAssignmentContainer/GET_STUDENT_DETAILS';
export const SAVE_ASSGINMENT_REQUEST = 'app/AssignmentContainer/SAVE_ASSGINMENT_REQUEST';
export const SAVE_ASSGINMENT_REQUEST_SUCCESS =
  'app/AssignmentContainer/SAVE_ASSGINMENT_REQUEST_SUCCESS';
export const SAVE_ASSGINMENT_REQUEST_ERROR =
  'app/AssignmentContainer/SAVE_ASSGINMENT_REQUEST_ERROR';
export const DELETE_ASSIGNMENT_REQUEST = 'app/AssignmentContainer/DELETE_ASSIGNMENT_REQUEST';
export const DELETE_ASSIGNMENT_REQUEST_SUCCESS =
  'app/AssignmentContainer/DELETE_ASSIGNMENT_REQUEST_SUCCESS';
export const DELETE_ASSIGNMENT_REQUEST_ERROR =
  'app/AssignmentContainer/DELETE_ASSIGNMENT_REQUEST_ERROR';
export const CLEAR_RESPONSE_STATUS = 'app/AssignmentContainer/CLEAR_RESPONSE_STATUS';
export const CLEAR_STATE = 'app/AssignmentContainer/CLEAR_STATE';
export const GET_STUDENTDETAIL_FAILURE = 'app/AssignmentContainer/GET_STUDENTDETAIL_FAILURE';
export const GET_ASSIGNMENT_FAILURE = 'app/AssignmentContainer/GET_ASSIGNMENT_FAILURE';
