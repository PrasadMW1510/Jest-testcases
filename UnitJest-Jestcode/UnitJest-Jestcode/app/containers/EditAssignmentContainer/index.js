/**
 *
 * EditAssignmentContainer
 *
 */

export { default } from './EditAssignmentContainer';
