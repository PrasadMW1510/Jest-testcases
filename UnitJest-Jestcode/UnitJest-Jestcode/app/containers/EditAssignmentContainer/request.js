import API from 'utils/request';

export const postAssignmentData = (sessionId, classId, requestData) =>
  API.post(`/SlmsSdp`, requestData, {
    params: {
      command: 'get_class_assignments',
      communityIds: 'R180NG,S44NG,S44JR,M180,M180Y2',
      cohort_id: classId,
      cohort_type: 'class',
      sid: sessionId,
    },
  }).then(response => response);

export const getStudentDetails = (sessionId, classId) =>
  API.post(`/SlmsClass`, '', {
    params: {
      command: 'get_enrollment',
      class_id: classId,
      sid: sessionId,
    },
  }).then(response => response);

export const saveAssignmentRequest = (sessionId, isStudentWorkItem, workItemId, payload) =>
  API.post(`/SlmsSdp`, payload, {
    params: {
      command: 'update_class_assignment',
      sid: sessionId,
      workItemId,
      isStudentWorkItem,
    },
  }).then(response => response);

export const deleteAssignmentRequest = (sessionId, isStudentWorkItem, workItemId) =>
  API.post(`/SlmsSdp`, '<>', {
    params: {
      command: 'delete_class_assignment',
      sid: sessionId,
      workItemId,
      isStudentWorkItem,
    },
  }).then(response => response);
