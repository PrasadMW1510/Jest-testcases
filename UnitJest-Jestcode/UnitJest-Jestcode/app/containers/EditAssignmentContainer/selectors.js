import { createSelector } from 'reselect';

/**
 * Direct selector to the editAssignmentContainer state domain
 */
const selectEditAssignmentContainerDomain = state => state.get('editAssignmentContainer');

/**
 * Other specific selectors
 */

/**
 * Default selector used by EditAssignmentContainer
 */

const makeSelectEditAssignmentContainer = () =>
  createSelector(selectEditAssignmentContainerDomain, substate => substate.toJS());

export default makeSelectEditAssignmentContainer;
export { selectEditAssignmentContainerDomain };
