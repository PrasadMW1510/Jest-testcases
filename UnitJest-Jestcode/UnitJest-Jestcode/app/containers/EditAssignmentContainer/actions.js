/*
 *
 * EditAssignmentContainer actions
 *
 */

import * as Constants from './constants';

export function getAssignmentData(data) {
  return {
    type: Constants.GET_ASSIGNMENT_REQUEST,
    data,
  };
}

export function getAssignmentDataSuccess(data) {
  return {
    type: Constants.GET_ASSIGNMENT_SUCCESS,
    data,
  };
}

export function getStudentDetails(data) {
  return {
    type: Constants.GET_STUDENT_DETAILS,
    data,
  };
}
export function getStudentDetailsSuccess(data) {
  return {
    type: Constants.GET_STUDENTDETAIL_SUCCESS,
    data,
  };
}

export function saveAssignmentRequest(data) {
  return {
    type: Constants.SAVE_ASSGINMENT_REQUEST,
    data,
  };
}

export function saveAssignmentRequestSuccess(data) {
  return {
    type: Constants.SAVE_ASSGINMENT_REQUEST_SUCCESS,
    data,
  };
}

export function saveAssignmentRequestError(error) {
  return {
    type: Constants.SAVE_ASSGINMENT_REQUEST_ERROR,
    error,
  };
}

export function deleteAssignmentRequest(data) {
  return {
    type: Constants.DELETE_ASSIGNMENT_REQUEST,
    data,
  };
}

export function deleteAssignmentRequestSuccess(data) {
  return {
    type: Constants.DELETE_ASSIGNMENT_REQUEST_SUCCESS,
    data,
  };
}

export function deleteAssignmentRequestError(error) {
  return {
    type: Constants.DELETE_ASSIGNMENT_REQUEST_ERROR,
    error,
  };
}

export function clearState() {
  return {
    type: Constants.CLEAR_STATE,
  };
}

export function clearResponseStatus() {
  return {
    type: Constants.CLEAR_RESPONSE_STATUS,
  };
}

export function getStudentDetailsFailure(error) {
  return {
    type: Constants.GET_STUDENTDETAIL_FAILURE,
    error,
  };
}

export function getAssignmentDataFailure(error) {
  return {
    type: Constants.GET_ASSIGNMENT_FAILURE,
    error,
  };
}
