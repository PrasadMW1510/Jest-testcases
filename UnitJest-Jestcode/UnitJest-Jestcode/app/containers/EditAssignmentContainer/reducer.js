/*
 *
 * EditAssignmentContainer reducer
 *
 */

import { fromJS } from 'immutable';
import * as Constants from './constants';

const initialState = fromJS({
  respAssignmentData: null,
  studentDetails: [],
  saveAssignmentSuccess: null,
  deleteAssignmentSucess: null,
});

function editAssignmentContainerReducer(state = initialState, action) {
  switch (action.type) {
    case Constants.GET_ASSIGNMENT_SUCCESS:
      return state.set(
        'respAssignmentData',
        fromJS(
          action.data.output_data[0].workItems[0].classAssignmentWorkItem[0].classAssignmentGroup[0]
        )
      );
    case Constants.GET_STUDENTDETAIL_SUCCESS:
      return state.set('studentDetails', action.data.output_data[0]);
    case Constants.SAVE_ASSGINMENT_REQUEST_SUCCESS:
      return state.set('saveAssignmentSuccess', action.data.output_data[0]);
    case Constants.DELETE_ASSIGNMENT_REQUEST_SUCCESS:
      return state.set('deleteAssignmentSucess', action.data.output_data[0]);
    case Constants.CLEAR_RESPONSE_STATUS:
      return state.set('saveAssignmentSuccess', null).set('deleteAssignmentSucess', null);
    case Constants.CLEAR_STATE:
      return state
        .set('studentDetails', [])
        .set('respAssignmentData', null)
        .set('saveAssignmentSuccess', null)
        .set('deleteAssignmentSucess', null);
    default:
      return state;
  }
}

export default editAssignmentContainerReducer;
