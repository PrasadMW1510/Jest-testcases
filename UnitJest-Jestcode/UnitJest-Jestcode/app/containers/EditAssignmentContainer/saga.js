import { call, all, put, select, takeLatest } from 'redux-saga/effects';
import * as Constants from './constants';
import { makeSelectProfileSessionId } from '../App/selectors';
import * as Actions from './actions';
import * as PortfolioActions from '../PortfolioPageContainer/actions';
import * as Request from './request';

export function* getAssignmentDataRequest(assignmentData) {
  try {
    const sessionId = yield select(makeSelectProfileSessionId());
    const requestData = `<workItemsSubset><workItemInfo workItemId="${
      assignmentData.data.classId.workItemId
    }" communityId= "${
      assignmentData.data.classId.workItemId.community_id
    }" studentId="null" assignment="${
      assignmentData.data.classId.assignment
    }" kind="ClassAssignment"/></workItemsSubset>`;
    const assignmentResultData = yield call(
      Request.postAssignmentData,
      sessionId,
      assignmentData.data.classId,
      requestData
    );
    yield put(Actions.getAssignmentDataSuccess(assignmentResultData));
    const studentDetails = yield call(
      Request.getStudentDetails,
      sessionId,
      assignmentData.data.classId.classId
    );

    yield put(Actions.getStudentDetailsSuccess(studentDetails));
  } catch (err) {
    yield put(Actions.getAssignmentDataFailure(err));
  }
}

export function* getStudentDetailsRequest(action) {
  try {
    const sessionId = yield select(makeSelectProfileSessionId());
    const classId = yield action.data;
    const studentDetails = yield call(Request.getStudentDetails, sessionId, classId);

    yield put(Actions.getStudentDetailsSuccess(studentDetails));
  } catch (err) {
    yield put(Actions.getStudentDetailsFailure(err));
  }
}

const generateSaveAssignmentPayload = data => {
  const { students, assignmentName, assignmentType, dueDate, description, communityId } = data;
  let tags;
  const studentsTags = students.map(student => {
    if (communityId === 'R180NG' || communityId === 'S44NG') {
      tags = `<student>
      <userId>${student.userId[0]}</userId>
      <average>${student.average[0]}</average>
      <comment>${student.comment[0]}</comment>
      <score>${student.score[0]}</score>
      <total>${student.total[0]}</total>
    </student>`;
    } else {
      tags = `<student>
                <userId>${student.userId[0]}</userId>
                <average>${student.average[0]}</average>
                <comment>${student.comment[0]}</comment>
              </student>`;
    }
    return tags;
  });
  const payload = `<classAssignmentGroup wholeClass="false">
      <assignmentName>${assignmentName}</assignmentName>
      <assignmentType>${assignmentType}</assignmentType>
      <dueDate>${dueDate}</dueDate> 
      <description>${description || ''}</description>
      <classAssignment communityId="${communityId}">
        <studentAssignments>
        ${studentsTags.join('\n')}
        </studentAssignments>
      </classAssignment>
    </classAssignmentGroup>`;
  return payload;
};

export function* saveAssignmentRequest(action) {
  const workItemId = yield action.data.metaData[action.data.currentIndex].workItemId;
  try {
    const sessionId = yield select(makeSelectProfileSessionId());
    const payload = yield generateSaveAssignmentPayload(action.data);
    const saveAssignmentResponse = yield call(
      Request.saveAssignmentRequest,
      sessionId,
      false,
      workItemId,
      payload
    );
    yield put(Actions.saveAssignmentRequestSuccess(saveAssignmentResponse));
    yield put(PortfolioActions.getStudentSubmissions());
  } catch (err) {
    yield put(Actions.saveAssignmentRequestError(err));
  }
}

export function* deleteAssignmentRequest(action) {
  const sessionId = yield select(makeSelectProfileSessionId());
  const { workItemId } = action.data;
  try {
    const deleteResponse = yield call(
      Request.deleteAssignmentRequest,
      sessionId,
      false,
      workItemId
    );
    yield put(Actions.deleteAssignmentRequestSuccess(deleteResponse));
  } catch (error) {
    yield put(Actions.deleteAssignmentRequestError(error));
  }
}

export default function* defaultSaga() {
  yield all([
    takeLatest(Constants.GET_ASSIGNMENT_REQUEST, getAssignmentDataRequest),
    takeLatest(Constants.GET_STUDENT_DETAILS, getStudentDetailsRequest),
    takeLatest(Constants.SAVE_ASSGINMENT_REQUEST, saveAssignmentRequest),
    takeLatest(Constants.DELETE_ASSIGNMENT_REQUEST, deleteAssignmentRequest),
  ]);
}
