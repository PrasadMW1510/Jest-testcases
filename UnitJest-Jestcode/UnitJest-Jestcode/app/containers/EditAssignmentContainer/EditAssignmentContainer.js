/**
 *
 * EditAssignmentContainer
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';
import { hideModal } from 'containers/ModalController/actions';
import EditAssignmentComponent from 'components/EditAssignmentComponent';
import injectSaga from 'utils/injectSaga';
import injectReducer from 'utils/injectReducer';
import makeSelectEditAssignmentContainer from './selectors';

import {
  getAssignmentData,
  getStudentDetails,
  saveAssignmentRequest,
  deleteAssignmentRequest,
  clearState,
  clearResponseStatus,
} from './actions';
import reducer from './reducer';
import saga from './saga';

export class EditAssignmentContainer extends React.Component {
  componentDidMount() {
    this.props.getAssignmentData(this.props.data);
    this.props.getStudentDetails(this.props.data.classId.classId);
  }

  handleCloseModal() {
    this.props.clearState();
    this.props.handleCancel();
  }

  render() {
    const { hideModal: hideModalProp } = this.props;
    return (
      <EditAssignmentComponent
        isOpen
        hideCancel={hideModalProp}
        closeStudentGoalModal={() => this.handleCloseModal()}
        data={this.props.editassignmentcontainer.respAssignmentData}
        {...this.props}
      />
    );
  }
}

EditAssignmentContainer.propTypes = {
  getAssignmentData: PropTypes.func.isRequired,
  data: PropTypes.object,
  editassignmentcontainer: PropTypes.object.isRequired,
  hideModal: PropTypes.func.isRequired,
  getStudentDetails: PropTypes.func.isRequired,
  clearState: PropTypes.func.isRequired,
  handleCancel: PropTypes.func.isRequired,
};

const mapStateToProps = createStructuredSelector({
  editassignmentcontainer: makeSelectEditAssignmentContainer(),
});

export function mapDispatchToProps(dispatch) {
  return {
    getAssignmentData: data => dispatch(getAssignmentData(data)),
    hideModal: () => dispatch(hideModal()),
    handleCancel: () => dispatch(hideModal()),
    getStudentDetails: data => dispatch(getStudentDetails(data)),
    saveAssignmentRequest: data => dispatch(saveAssignmentRequest(data)),
    deleteAssignmentRequest: data => dispatch(deleteAssignmentRequest(data)),
    clearState: () => dispatch(clearState()),
    clearResponseStatus: () => dispatch(clearResponseStatus()),
  };
}

const withConnect = connect(mapStateToProps, mapDispatchToProps);

const withReducer = injectReducer({ key: 'editAssignmentContainer', reducer });
const withSaga = injectSaga({ key: 'editAssignmentContainer', saga });

export default compose(withReducer, withSaga, withConnect)(EditAssignmentContainer);
