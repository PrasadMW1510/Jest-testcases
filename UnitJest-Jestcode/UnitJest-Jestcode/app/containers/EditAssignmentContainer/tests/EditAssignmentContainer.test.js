import React from 'react';
import { shallowToJson } from 'enzyme-to-json';
import { shallow } from 'enzyme';
import { EditAssignmentContainer, mapDispatchToProps } from '../EditAssignmentContainer';

describe('EditAssignmentContainer ', () => {
  let wrapper = null;
  const mockdata = {
    classId: {
      classId: 'fff',
    },
  };
  const mockeditassignmentcontainer = {};
  const mockgetStudentDetails = jest.fn();
  beforeEach(() => {
    wrapper = shallow(
      <EditAssignmentContainer
        saveAssignmentRequest={jest.fn()}
        deleteAssignmentRequest={jest.fn()}
        clearResponseStatus={jest.fn()}
        getAssignmentData={jest.fn()}
        hideModal={jest.fn()}
        data={mockdata}
        editassignmentcontainer={mockeditassignmentcontainer}
        getStudentDetails={mockgetStudentDetails}
        clearState={jest.fn()}
        handleCancel={jest.fn()}
      />
    );
  });
  it('Should render TeacherMadeQuizContainer', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should call handleCloseModal', () => {
    wrapper.instance().handleCloseModal();
    expect(wrapper.instance().props.handleCancel).toBeCalled();
  });

  it('map Dispatch To Props', () => {
    const mockObject = jest.fn;
    const mapDispatchToPropscomp = mapDispatchToProps(mockObject);
    expect(mapDispatchToPropscomp.getAssignmentData().mock.calls.length).toBe(0);
    expect(mapDispatchToPropscomp.hideModal().mock.calls.length).toBe(0);
    expect(mapDispatchToPropscomp.getStudentDetails().mock.calls.length).toBe(0);
    expect(mapDispatchToPropscomp.saveAssignmentRequest().mock.calls.length).toBe(0);
    expect(mapDispatchToPropscomp.clearResponseStatus().mock.calls.length).toBe(0);
    expect(mapDispatchToPropscomp.clearState().mock.calls.length).toBe(0);
    expect(mapDispatchToPropscomp.deleteAssignmentRequest().mock.calls.length).toBe(0);
    expect(mapDispatchToPropscomp.handleCancel().mock.calls.length).toBe(0);
  });
});
