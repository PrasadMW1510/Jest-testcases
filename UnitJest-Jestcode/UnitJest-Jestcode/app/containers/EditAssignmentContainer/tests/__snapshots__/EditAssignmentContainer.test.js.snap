// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditAssignmentContainer  Should render TeacherMadeQuizContainer 1`] = `
<EditAssignmentComponent
  clearResponseStatus={[Function]}
  clearState={[Function]}
  closeStudentGoalModal={[Function]}
  data={
    Object {
      "classId": Object {
        "classId": "fff",
      },
    }
  }
  deleteAssignmentRequest={[Function]}
  editassignmentcontainer={Object {}}
  getAssignmentData={[Function]}
  getStudentDetails={[Function]}
  handleCancel={[Function]}
  hideCancel={[Function]}
  hideModal={[Function]}
  isOpen={true}
  saveAssignmentRequest={[Function]}
/>
`;
